client/landing/gutenboarding/components/domain-picker/button.tsx

```diff
@@ -4,27 +4,70 @@
 import React, { FunctionComponent, useState } from 'react';
 import { __ as NO__ } from '@wordpress/i18n';
 import { Button, Popover } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import DomainPicker from './list';
+import { STORE_KEY as DOMAIN_STORE } from '../../stores/domain-suggestions';
+import { STORE_KEY as ONBOARD_STORE } from '../../stores/onboard';
+import { isFilledFormValue } from '../../stores/onboard/types';
+
+import './style.scss';
 
 const DomainPickerButton: FunctionComponent = () => {
+	// // User can search for a domain
+	const [ domainSearch, setDomainSearch ] = useState( '' );
+
 	const [ isDomainPopoverVisible, setDomainPopoverVisibility ] = useState(
 		true /* @TODO: should be `false` by default, true for dev */
 	);
 
+	// Without user search, we can provide recommendations based on title + vertical
+	const { siteTitle, siteVertical } = useSelect( select => select( ONBOARD_STORE ).getState() );
+
+	let search = domainSearch.trim();
+	if ( ! search && isFilledFormValue( siteTitle ) ) {
+		search = siteTitle;
+	}
+
+	const suggestions = useSelect(
+		select => {
+			if ( search ) {
+				return select( DOMAIN_STORE ).getDomainSuggestions( search, {
+					include_wordpressdotcom: true,
+					...( isFilledFormValue( siteVertical ) && { vertical: siteVertical.id } ),
+				} );
+			}
+		},
+		[ search, siteVertical ]
+	);
+
+	// Free .wordpress.com subdomain at index 0,
+	// Best paid domain match at index 1.
+	const [ freeDomainSuggestion, ...paidDomainSuggestions ] = suggestions ?? [];
+
 	return (
-		<Button onClick={ () => setDomainPopoverVisibility( s => ! s ) }>
-			{ NO__( 'Pick a domain' ) }
+		<Button
+			className="domain-picker__button"
+			onClick={ () => setDomainPopoverVisibility( s => ! s ) }
+		>
+			<div className="domain-picker__site-title">
+				{ siteTitle ? siteTitle : NO__( 'Create your site' ) }
+			</div>
```

Comment:
moving Site title here means it will be also clickable to trigger the domain picker but looking at the designs (both v1 and v2), since the arrow is placed next to the domain, as a user, I would expect just the domain to be clickable.

Reply:
We'll need to rework this structure, but let's land this then iterate.

[`button`](https://html.spec.whatwg.org/dev/form-elements.html#the-button-element) should have non-interactive [phrasing content](https://html.spec.whatwg.org/dev/dom.html#phrasing-content). `div` children and the `Popover` all violate this 😞

Reply:
> We'll need to rework this structure, but let's land this then iterate.
> 
> [`button`](https://html.spec.whatwg.org/dev/form-elements.html#the-button-element) should have non-interactive [phrasing content](https://html.spec.whatwg.org/dev/dom.html#phrasing-content). `div` children and the `Popover` all violate this

Ugh, I tried to avoid a11y violations by keeping this a `Button`, rather than turning it into a `div`, which would then need an ARIA role (easy), and, (harder), a key handler -- while `Button` probably gives us all that for free. But yeah, apparently I couldn't work my way around it totally.

Reply:
Let's not worry too much about that here, we can move things around. This includes some nice state changes 👍